### PR TITLE
Rdm 9113 1

### PR DESCRIPTION
**Pull request by skandala-cgi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-9113

### Change description ###

Migrate DataStore to use HMCTS Gradle Java Plugin

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by MSancaktutar — APPROVED** [review]
LGTM but better to get a second review.

@banderous  - Could you pls give this a quick glance? This is a huge change and it's important to make sure it lands correctly.

File: build.gradle [hunk 1]
```diff
@@ -142,25 +132,19 @@ configurations {
     cucumberRuntime.extendsFrom(functionalRuntime)
 }
 
-checkstyle {
-    showViolations = true
-    maxWarnings = 0
-    toolVersion = '8.18'
-}
+//checkstyleMain {
```
Comment (banderous) [inline build.gradle:135]: should this be here?

File: build.gradle [hunk 2]
```diff
@@ -44,18 +42,19 @@ apply plugin: 'io.spring.dependency-management'
...
 apply plugin: 'com.jfrog.bintray'
 apply plugin: 'se.patrikerdes.use-latest-versions'
-apply plugin: 'org.owasp.dependencycheck'
 
 apply from: 'liquibase.gradle'
 
 pmd {
-    toolVersion = '6.11.0'
-    maxFailures = 3150 // Failures need to be reviewed/fixed before removing this property
+    toolVersion = '6.18.0'
```
Comment (banderous) [inline build.gradle:53]: PMD is not used or required by the gradle plugin, is this change coincidental?

**Review by banderous — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (MSancaktutar) [reply build.gradle:135]: Well, it looks like it's just missed.
@skandala-cgi, could you please remove checkstyle plugin traces entirely?

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
Please remove all traces of Checkstyle plugin, and for now retain PMD.
PMD will be removed under a separate ticket from other apps altogether in a consistent plan.

File: build.gradle [hunk 3]
```diff
@@ -142,24 +132,9 @@ configurations {
...
-    reports {
-        xml.enabled = false
-        html.enabled = true
-    }
-}
-
-spotbugs {
-    ignoreFailures = true
+//This is to exclude generated files from build/generated-sources/js2p/uk/gov/hmcts/ccd/domain/types/sanitiser/document
+tasks.withType(Checkstyle) {
```
Comment (MSancaktutar) [inline build.gradle:136]: wouldn't this be removed as well?

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
still some trace of checkstyle perhaps

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Comment by VamshiRudrabhatlaHMCTS** [comment]
This ticket is verified and tested successfully. The check styles were successfully verified from the logs as the part of new HMCTS Java Gradle plug in. Sanity tests (CCD-UI) satisfied. Happy to merge this ticket.

**Pull request merged**